Standish Code

Granto1 opened the pull request:
Everything tested on 3/15.
Top-Auton and Bottom-Auton need to be re-tested on Standish playing field. Need to fix the waypoints there.

boboqiu05 committed:
update

boboqiu05 committed:
update

Granto1 committed:
3/1 Before SysID

Added ticksPerMeters in Constants for Encoders

boboqiu05 committed:
update

Tyler-Bryant committed:
Constants are not correct numbers, but basic logic for implementing SysID has begun

Tyler-Bryant committed:
Implemented code created from plybot to run different paths. Will help to test SysID numbers

boboqiu05 committed:
update

boboqiu05 committed:
saturday morning test

sahibakaur771 committed:
neutral deadband for joysticks in initializeMotors method

boboqiu05 committed:
sat afternoon

boboqiu05 committed:
code with detailed description to make easier to reuse in new code

Granto1 committed:
Shifting to Prog 7

Tyler-Bryant committed:
Configured for SysID testing and added old paths

Granto1 committed:
3/8 Pre-Testing

Comments, poked at some stuff

Granto1 committed:
More Tinkering

Tyler-Bryant committed:
Added comments and put in values that have been tuned. We should tune the values more in the future to go faster accurately.

Granto1 committed:
3/8 BangBang Controller "works" (180 overshoot, 90 undershoot)

Need to spend time on MotionMagic.

boboqiu05 committed:
no more armstate

boboqiu05 committed:
boboarm2

Granto1 committed:
Formatting, Correct Yaml, RobotBuilder Integration

boboqiu05 committed:
push

Granto1 committed:
3/9 SafetyCommand Functions

Not sure why the Arm code on this branch doesn't work...

Granto1 committed:
Merge branch 'BoboArm2' into WorkingTestBranch

Granto1 committed:
Merge branch 'ClawSafetyCommand' into WorkingTestBranch

Granto1 committed:
3/9 Creation of Testing Branch

Formatting

Granto1 committed:
3/10 Clean Up

Deleted Unnecessary Arm, Some Comments, Swapped Joystick for MoveArmSimple to remove Joystick Conflicts

Tyler-Bryant committed:
adds brake and coast modes when the robot is enabled/disabled

Tyler-Bryant committed:
Removed unneccessary comments and increaced speed of robot. It now goes 3 m/s and 2m/s^2 max

Granto1 committed:
3/10 Camera "Works" + More Changes to Come

ClawSafety needs a larger bound...
Removed some un-needed arm code

Granto1 committed:
Merge remote-tracking branch 'origin/SahibaDeadband' into WorkingTestBranch

boboqiu05 committed:
Merged BoboArm2 into WorkingTestBranch

Granto1 committed:
3/11 Some paths to test today

Tyler-Bryant committed:
Prep for robot builder. Lets hope this works.

Tyler-Bryant committed:
Part of the Yaml is completed

Tyler-Bryant committed:
Merge branch 'WorkingTestBranch' of https://github.com/The-Charge/ChargedUp2023 into WorkingTestBranch

boboqiu05 committed:
added climb code

Granto1 committed:
3/11 Quick commit to bring back stashed changes

Granto1 committed:
3/11 Prepared Velocity Control for Testing

Granto1 committed:
Merge branch 'BoboWorkingTestBranch' into WorkingTestBranch

Tyler-Bryant committed:
All paths in TylerSysID branch are now designated a bucket in the m_pathSendableChooser. In short, new paths are runnable.

Granto1 committed:
3/11 Climbs Work, Arm Positions tinkered with

Tyler-Bryant committed:
Framework for event map has been made, some paths have been changed to reduce the angle the robot drives at.

Tyler-Bryant committed:
Update to pathplanner. It was stashed for some reason, so I have to make this a seperate commit

Granto1 committed:
Merge branch 'WorkingTestBranch' into TylerSysId

Granto1 committed:
Merge branch 'TylerSysId' of https://github.com/The-Charge/ChargedUp2023 into TylerSysId

Granto1 committed:
3/12 HashMap Events

boboqiu05 committed:
auton

Granto1 committed:
3/12 Tested PID, still need to tune

Waiting on thread to be reinstalled

Granto1 committed:
3/12 Tinkered with PID

Granto1 committed:
3/12 Tested capability to add events on paths. Can do

Need to add events on multiple paths

Granto1 committed:
3/12 Worked with Align. Distance is fine but angle is off

Granto1 committed:
Merge remote-tracking branch 'origin/BoboTestBranch2' into WorkingTestBranch

Granto1 committed:
3/13 SetBrake() in Tele-Init

Granto1 committed:
Merge remote-tracking branch 'origin/BoboTestBranch2' into TylerSysId

Granto1 committed:
3/13 Removed Un-needed Path and Updated AutonCommand List

boboqiu05 committed:
updated can ids and shoulder encoder

Tyler-Bryant committed:
Merge remote-tracking branch 'origin/BoboTestBranch2' into TylerSysId

Tyler-Bryant committed:
Merge branch 'TylerSysId' of https://github.com/The-Charge/ChargedUp2023 into TylerSysId

boboqiu05 committed:
updated buttons

Tyler-Bryant committed:
integrated 5 cases where a pre-made Sequential Command Group would be run instead of a path

Granto1 committed:
3/13 Changed Shlder Absolute Encoder Offset and sensorphase, need to figure out why shoulder movement is inverted

Granto1 committed:
Merge remote-tracking branch 'origin/BoboTestBranch2' into WorkingTestBranch

Granto1 committed:
Merge remote-tracking branch 'origin/BoboTestBranch2' into TylerSysId

Granto1 committed:
3/14 Updated Middle SequentialCommands to reflect parameters from the climb/drive over commands

Granto1 committed:
3/14 Rebuilt ClawSafety

Checking if within robot bounds now occurs by sending isArmTipinsideRobotX to Claw Subsystem to determine whether to open or close.

abcurrie committed:
Reduced Shoulder Motor Gain

Gain was too high

abcurrie committed:
Set Shldr to Coast when disables

abcurrie committed:
Changed POV code

Cj

Granto1 committed:
3/14 Force Claw Open for Sequential Command

Granto1 committed:
3/14 Still in progress

Everything leading up to the 180 turn works

Tyler-Bryant committed:
Formating, changed amount of paths from 20 to 17, and changed array name of temp path storer

Granto1 committed:
Merge branch 'WorkingTestBranch' into TylerSysId

Tyler-Bryant committed:
some unused functions were not removed on the last commit.

Granto1 committed:
Merge branch 'TylerSysId' of https://github.com/The-Charge/ChargedUp2023 into TylerSysId

Granto1 committed:
Merge branch 'TylerSysId' of https://github.com/The-Charge/ChargedUp2023 into TylerSysId

Granto1 committed:
3/14 New Path, needs testing

boboqiu05 committed:
3/14 Corrected Auton Heading

boboqiu05 committed:
3/14 Merged TylerSysID into WorkingTestBranch

Granto1 committed:
3/14 Last Working Branch

Need to clean up before pull request

Granto1 committed:
Merge branch 'WorkingTestBranch' of https://github.com/The-Charge/ChargedUp2023 into WorkingTestBranch

boboqiu05 committed:
score on high goal

boboqiu05 committed:
update

Tyler-Bryant committed:
Formatted the Working Test Branch so that the code is readable to the average programmer. Feel free to clean it up more, but I got the main highlights; commenting etiquette, one parentheses around a confusing equation, etc.

Granto1 committed:
3/15 Pre-Standish Final Commit

Notes will be in Pull Request

boboqiu05 committed:
To allow Z for arms.  Changed line 261 in MagicArm

ammenzl7 commented:
Were the TestBranchYaml changes incorporated into this branch? I see there was another pull request created to merge them into this branch.

ammenzl7 commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -21,77 +22,113 @@
+    // SmartDashboard.putData("CloseClaw", new CloseClaw(m_claw));
+    // SmartDashboard.putData("ReverseDrive", new ReverseDrive(m_drivetrain));
+    // SmartDashboard.putData("HalfSpeed", new HalfSpeed(m_drivetrain));
+    // SmartDashboard.putData("QuarterSpeed", new QuarterSpeed(m_drivetrain));
```
These dashboard values are commented in autogenerated code. The next time robotbuilder is ran, the comments will be removed. If these aren't needed anymore, remove them from robotbuilder, run robotbuilder and format all the files please.

ammenzl7 commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -100,52 +137,151 @@ public static RobotContainer getInstance() {
+
+    // final JoystickButton turnNDegrees180Btn = new JoystickButton(rightJoystick, 12);
+    // turnNDegrees180Btn
+    //     .onTrue(new TurnNRelative(m_drivetrain, 180).withInterruptBehavior(InterruptionBehavior.kCancelSelf));
```
Same with all these. If things aren't needed from robotbuilder, remove them from robotbuilder instead of commenting them.

ammenzl7 commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -100,52 +137,151 @@ public static RobotContainer getInstance() {
+     * JoystickButton backHighButton = new JoystickButton(armController, 2);
+     * backHighButton.onTrue(armBackHigh);
+     */
+    // SmartDashboard.putData("AutoSelect", m_chooser);
```
See previous comments

ammenzl7 commented on src/main/java/frc/robot/commands/AutonomousCommand.java:
```diff
@@ -12,40 +12,199 @@
     // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CONSTRUCTORS
 
-    public AutonomousCommand() {
+    public AutonomousCommand(Drivetrain subsystem, MagicArm subsystem2, Claw subsystem3) {
```
I don't think robotbuilder can possibly add this change, so if this is manual, make sure robotbuilder doesn't include the autonomous command in it's scope. Otherwise all of this will be overridden.

ammenzl7 commented on src/main/java/frc/robot/commands/AutonomousCommand.java:
```diff
@@ -12,40 +12,199 @@
+                return scoreHighConeCommand();
+            case "Score High Ball":
+                return climbCommand();
+            default:
```
The contents of the default case should really be in a separate method

ammenzl7 commented on src/main/java/frc/robot/subsystems/Drivetrain.java:
```diff
@@ -12,121 +12,142 @@
     rightRearMotor.follow(rightFrontMotor);
     leftRearMotor.follow(leftFrontMotor);
 
+    setBrake();
```
This is already called 5 lines before this, is one of them redundant?

Granto1 committed:
3/16 Adjustment of Paths

Needed additional markers for movement to neutral.
Test on field.

Granto1 committed:
3/16 Standish Adjustments

Increased DriveOver Ticks
Decreased "Dunk" distance
Less climbPower
Adjustment of substation x as well as increase of y limit

Granto1 committed:
3/16 Path Adjustment

Removed movement to intermediate point in favor of immediate movement towards neutral once the robot leaves the scoring node zone.

Granto1 committed:
3/17 Standish Adjustments

Everything going well

Tyler-Bryant committed:
Test branch yaml (#4)

* Fixed button names to be more accurate to what they do.

* Robot builder has been formated. I will be doing robot builder again to double check changes are sound

* Second time robot builder has been pushed to code. I'm going to do it one more time to make sure things aren't broken.

* Third iteration. I'm going to do another pass through after lunch

* fourth iteration of robot builder. Requires a fifth run through

* It works. There are two commands; Align and claw safety; and one button; AlignBtn; that are now removed due to creation of errors.

* Formatting is complete. Mr. Menzl should look to see if it is good. Main thing is the @param and @return comments. It doesn't seem right to me. Thank you Bobo for this.

* Fixed issues presented by Mr. Menzl

* Fixed errors that were caused by merging WorkingTestBranch. Needs testing on the robot.

* removed warnings and fixed Mr. Menzl's nit picks.

* Fixed Mr. Menzl's code rejects he pointed out

* fixed some things in code (brake mode etc.)

---------

Co-authored-by: Granto1 <63463218+Granto1@users.noreply.github.com>

Granto1 committed:
3/20 Fixed errors from pull request #4

boboqiu05 committed:
calculations are made in radians and meters. Constants are in inches

boboqiu05 committed:
Merge branch 'WorkingTestBranch' of https://github.com/The-Charge/ChargedUp2023 into WorkingTestBranch

Granto1 committed:
3/20 Check Desc

Cleaned up Auton paths: removed unnecessary paths. Added clear/bump paths without events and renamed chooser list.
Formatting adjusted.

ammenzl7 commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -40,26 +40,27 @@ public static final class AutoConstants {
-    public static final double shoulderL = 0.9906;
-    public static final double elbowL = 0.9144;
-    public static final double shoulderAngleToSafeSwingElbowThrough = 0.395;
+    public static final double shoulderL = 39.0 * 0.0254;
```
In these kind of situations, constants are needed in case these kind of multipliers need to be updated so none are forgotten. 0.0254 should be a constant. Same with PI

ammenzl7 replied:
@boboqiu05

ammenzl7 commented:
I'd like to see the robot running with this code before merging.

boboqiu05 committed:
more comments

boboqiu05 committed:
Merge branch 'WorkingTestBranch' of https://github.com/The-Charge/ChargedUp2023 into WorkingTestBranch

boboqiu05 committed:
conversion constant added

boboqiu05 committed:
raise ground limit from 0.02 to 0.04

Granto1 committed:
3/21 Quick changes before pulling Bobo's Changes

Will document later

Granto1 committed:
Merge branch 'WorkingTestBranch' of https://github.com/The-Charge/ChargedUp2023 into WorkingTestBranch

Granto1 committed:
Revert "Merge branch 'WorkingTestBranch' of https://github.com/The-Charge/ChargedUp2023 into WorkingTestBranch"

This reverts commit 67946c6825e5cac1627a10e095e124e402a3decc, reversing
changes made to ebf841ed19de0c6e18c8e592548d66a8cda7a3f0.

boboqiu05 committed:
Fixed angular conversion

boboqiu05 committed:
ScoreHighConeChargeStation and high XY adjusted

Granto1 committed:
Merge branch 'WorkingTestBranch' of https://github.com/The-Charge/ChargedUp2023 into WorkingTestBranch

Granto1 committed:
3/21 Elbow is fast now

Doubled elbow speed, decreased gain.
High to high point is now roughly 4 seconds compared to 7 seconds previously

boboqiu05 committed:
3/22 Adjustments to arm movement before elbow motor fried

boboqiu05 committed:
switching over to 7

Granto1 committed:
3/22 Pre-Midland Commit

Removal of unnecessary auton paths.
Cleaned up auton path names.
Commented out all unused vision code.
Adjustment of mid/substation positions.

Granto1 committed:
3/23 Fixed Auton Names

Granto1 committed:
3/24 Midland Day One

TODO
Add HeadingOffsets for ChargeStationTwoBall then test in game.
Ask folk about pathplanner stop event

Changes:
Multiple auton failures. Added timeout for DriveOver.
DriveOverDistance added for TwoBall
Adjustment of Station coordinates.

ammenzl7 commented:
Merge pull request when ready, don't squash and merge

Granto1 committed:
3/25 Midland Day Two

Added Two Ball Charge Station Path (Picks up a piece after first score) that needs testing on red side.
Adjusted Climb after a near balance fail.
